openff/toolkit/topology/topology.py

```diff
@@ -2385,3 +2386,25 @@ def hierarchy_iterator(
             if hasattr(molecule, iter_name):
                 for item in getattr(molecule, iter_name):
                     yield item
+
+    def __getattr__(self, name: str) -> List["HierarchyElement"]:
+        """If a requested attribute is not found, check the hierarchy schemes"""
+        # Avoid attempting to process dunder methods as hierarchy scheme iterator names
+        if name.startswith("__"):
+            raise AttributeError
+
+        if name in dir(self):
+            raise HierarchyIteratorNameConflictError(
+                f"Name {name} is already defined as an attribute/method of `Topology`."
+            )
```

Comment:
This was meant to be something of a double safeguard even though it can't be reached in the current implementation. When there is a clash, I'm not sure which behavior from `Topology` the user would get.